web/files--common/javascript/OZONE.js

```diff
@@ -1,767 +1,730 @@
+/* global YAHOO, $, fx */
 
+const ogettext = function (mid) {
+  return OZONE.loc.getMessage(mid, OZONE.lang)
+}
 
-var OZONE = function(){}
-
-OZONE.ajax = {
-
-	_callbackArray: new Array(),
-	_callbackArrayIndex: 0,
-	_javascriptLoadLock: false,
-
-	/**
-	 * arg - extra parameter passed to the callback as a second parameter
-	 */
-	requestModule: function(moduleName, parameters, callback, arg, options){
-		OZONE.visuals.cursorWait();
-		if(parameters==null){
-			parameters = new Object();
-		}
-		if(moduleName == null || moduleName == ""){
-			moduleName = "Empty";
-		}
-		parameters["moduleName"] = moduleName;
-
-		if(options && options.clearRequestQueue){
-			OZONE.ajax._callbackArray = new Array();
-		}
-
-		var callbackIndex = OZONE.ajax._callbackArrayIndex++;
-		OZONE.ajax._callbackArray[callbackIndex] = {callback: callback, arg: arg};
-
-		parameters['callbackIndex'] = callbackIndex;
-
-		// add token information
-		var token = OZONE.utils.getCookie("wikijump_token7");
-		if(token == null){
-			alert("Error processing the request.\n\n" +
-					"You have no valid security token which is required to prevent " +
-					"identity theft.\n" +
-					"Please enable cookies " +
-					"in your browser if you have this option disabled " +
-					"and reload the page.");
-			OZONE.visuals.cursorClear();
-			return;
-		}
-		parameters['wikijump_token7'] = token;
-
-		var postdata = OZONE.utils.arrayToPostData(parameters);
-		var internalCallback = OZONE.ajax.requestModuleCallback;
-		YAHOO.util.Connect.asyncRequest('POST','/ajax-module-connector.php',internalCallback,postdata);
-
-	},
-
-	parseResponse: function(jsonResponse){
-		res = JSON.parse(jsonResponse);
-        if(!res){
-        	alert(jsonResponse.replace(/\r?\n/g,' '));
+const OZONE = {
+
+  ajax: {
+
+    _callbackArray: [],
+    _callbackArrayIndex: 0,
+    _javascriptLoadLock: false,
+
+    /**
+     * arg - extra parameter passed to the callback as a second parameter
+     */
+    requestModule: function (moduleName, parameters, callback, arg, options) {
+      OZONE.visuals.cursorWait()
+      if (parameters === null) {
+        parameters = {}
```

Comment:
This is something we need to discuss, then. I'm using [JavaScript Standard Style] (https://standardjs.com/), which is a linter and fixer that enforces a 'standard' style and crucially does not have any configuration. The point of it is to prevent wasting time on this sort of discussion. 

However, standard doesn't work with Typescript out of the box and its own FAQ recommends you replace it with [standardx](https://github.com/standard/standardx), a fork of it that also enables it to be configured. So while it's against the ideology of standard, it is possible to enable semicolons. 

It's very rare that a semicolon in JS is syntactically necessary. I'd strongly prefer to go without - if we're writing code that needs semicolons in order not to be ambiguous, whether that's syntactically or visually, we're writing bad code. 

Equally, though, I wouldn't merge a commit that removes semicolons without at least looking through the code first.